File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
  def isPalindrome(self, x: int) -> bool:
    if x < 0: return False
    if x < 10: return True
    
    s = str(x)
    l, r = 0, len(s)-1
    while l < r:
      if s[l] != s[r]:
        return False
      else:
        l += 1
        r -= 1
    return True

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
  public boolean isPalindrome(int x) {
    if (x < 0) return false;
    if (x < 10) return true;
    
    String s = Integer.toString(x);
    int l = 0;
    int r = s.length() - 1;
    while (l < r) {
      if (s.charAt(l) != s.charAt(r)) {
        return false;
      } else {
        l++;
        r--;
      }
    }
    return true;
  }
}